### ISSUE TITLE
Trying to publish from a subfolder wrongfully looks to the wrong folder

### ISSUE THREAD

**Issue by XhmikosR** [netlify/cli#5948]
### Describe the bug

I'm trying to publish a test site build for https://github.com/GoogleChromeLabs/quicklink.

As you can see, the website code is in the `site` folder, and both the root folder and the `site` subfolder contain a package.json and a .gitignore file.

```
C:\Users\xmr\Desktop\quicklink>netlify --version
netlify-cli/16.0.2 win32-x64 node-v18.17.1

C:\Users\xmr\Desktop\quicklink>cd site

C:\Users\xmr\Desktop\quicklink\site>npm ci

added 426 packages, and audited 427 packages in 2s

98 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

C:\Users\xmr\Desktop\quicklink\site>npm run build

> eleventy-quicklink-website@1.0.0 build
> rimraf build && cross-env NODE_ENV=production eleventy

[11ty] Writing build/robots.txt from ./src/robots.njk
[11ty] Writing build/assets/styles/copy-snippet-8016e282.css from ./src/assets/styles/copy-snippet.scss
[11ty] Writing build/sitemap.xml from ./src/sitemap.njk
[11ty] Writing build/assets/styles/github-markdown-56a0761c.css from ./src/assets/styles/github-markdown.scss
[11ty] Writing build/assets/styles/vendor/prism-b9e16d48.css from ./src/assets/styles/vendor/prism.scss
[11ty] Writing build/assets/styles/main-0badfe4e.css from ./src/assets/styles/main.scss
[11ty] Writing build/approach/index.html from ./src/approach.njk
[11ty] Writing build/demo/index.html from ./src/demo.njk
[11ty] Writing build/measure/index.html from ./src/measure.njk
[11ty] Writing build/api/index.html from ./src/api.njk
[11ty] Writing build/index.html from ./src/index.njk
[11ty] Copied 44 files / Wrote 11 files in 0.52 seconds (47.3ms each, v2.0.1)

C:\Users\xmr\Desktop\quicklink\site>netlify deploy -d build --prod
This folder isn't linked to a site yet
? What would you like to do? +  Create & configure a new site
? Team: xhmikosr’s team
? Site name (leave blank for a random name; you can change it later):

Site Created

Admin URL: https://app.netlify.com/sites/hilarious-sable-cee311
URL:       https://hilarious-sable-cee311.netlify.app
Site ID:   d5f82ede-fe4f-4e8d-bea5-180c3bc51c90


Adding local .netlify folder to .gitignore file...
Linked to hilarious-sable-cee311

Deploy path: C:\Users\xmr\Desktop\quicklink\build
 »   Error: No such directory C:\Users\xmr\Desktop\quicklink\build! Did you forget to run a build?

```

### Steps to reproduce

1. `git clone https://github.com/GoogleChromeLabs/quicklink.git`
2. `cd site`
3. `npm ci`
4. `npm run build`
5. `netlify deploy -d build --prod`

### Environment

```
netlify-cli/16.0.2 win32-x64 node-v18.17.1

System:
  OS: Windows 10 10.0.19045
  CPU: (8) x64 13th Gen Intel(R) Core(TM) i7-13700K
  Memory: 5.54 GB / 8.00 GB
Binaries:
  Node: 18.17.1 - C:\Program Files\nodejs\node.EXE
  npm: 9.6.7 - C:\Program Files\nodejs\npm.CMD
```

---

I would expect this to work, am I missing something?

EDIT: it seems that deploying from the root folder works (`netlify deploy -d site/build --prod`), but shouldn't this be a little clearer? Like output a warning, or even better deploy from the subfolder and not the root folder.

**Comment by CamiloTerevinto** [comment]
I have a GitHub action that does `cd ./out && netlify deploy --dir .`. This gave me a log of:

> Deploy path: /home/runner/work/www/www/out

Now, it's giving a log of:

> Deploy path: /home/runner/work/www/www

Please fix this

**Comment by dfdeagle47** [comment]
@XhmikosR It might be the same issue as https://github.com/netlify/cli/issues/5949. AFAIK, there was a breaking change because, at least for me, I could use a path relative to the cwd instead of a path relative to the root of the repo folder.

**Comment by lukasholzer** [comment]
@XhmikosR this got fixed in the latest release [16.3.3](https://github.com/netlify/cli/commit/da546769a93394a00b20a158c93b1e778102873d)
